tests/test_orchestration_executor.py

```diff
@@ -228,6 +228,173 @@ def orchestrator(ctx: task.OrchestrationContext, orchestrator_input):
     assert user_code_statement in complete_action.failureDetails.stackTrace.value
 
 
+def test_activity_retry_policies():
+    """Tests the retry policy logic for activity tasks"""
+
+    def dummy_activity(ctx, _):
+        raise ValueError("Kah-BOOOOM!!!")
+
+    def orchestrator(ctx: task.OrchestrationContext, orchestrator_input):
+        result = yield ctx.call_activity(
+            dummy_activity,
+            retry_policy=task.RetryPolicy(
+                first_retry_interval=timedelta(seconds=1),
+                max_number_of_attempts=6,
+                backoff_coefficient=2,
+                max_retry_interval=timedelta(seconds=10),
+                retry_timeout=timedelta(seconds=50)),
+            input=orchestrator_input)
+        return result
+
+    registry = worker._Registry()
+    name = registry.add_orchestrator(orchestrator)
+
+    current_timestamp = datetime.utcnow()
+    # Simulate the task failing for the first time and confirm that a timer is scheduled for 1 second in the future
+    old_events = [
+        helpers.new_orchestrator_started_event(timestamp=current_timestamp),
+        helpers.new_execution_started_event(name, TEST_INSTANCE_ID, encoded_input=None),
+        helpers.new_task_scheduled_event(1, task.get_name(dummy_activity))]
+    expected_fire_at = current_timestamp + timedelta(seconds=1)
+    
+    new_events = [
+        helpers.new_orchestrator_started_event(timestamp=current_timestamp),
+        helpers.new_task_failed_event(1, Exception("Kah-BOOOOM!!!"))]
+    executor = worker._OrchestrationExecutor(registry, TEST_LOGGER)
+    actions = executor.execute(TEST_INSTANCE_ID, old_events, new_events)
+    print(actions)
```

Comment:
Should we remove these `print(actions)` statements? I assume those were used for debugging?

Reply:
my bad, I have removed